cirq/google/calibration/workflow.py

```diff
@@ -14,24 +14,50 @@
 from typing import Callable, Dict, List, Optional, Sequence, Tuple, Union, cast
 
 import dataclasses
+from itertools import zip_longest
 
 
 from cirq.circuits import Circuit
-from cirq.ops import FSimGate, Gate, GateOperation, MeasurementGate, Moment, Qid, SingleQubitGate
+from cirq.ops import (
+    FSimGate,
+    Gate,
+    GateOperation,
+    MeasurementGate,
+    Moment,
+    Qid,
+    SingleQubitGate,
+    rz,
+)
 from cirq.google.calibration.engine_simulator import PhasedFSimEngineSimulator
 from cirq.google.calibration.phased_fsim import (
     FloquetPhasedFSimCalibrationOptions,
     FloquetPhasedFSimCalibrationRequest,
     IncompatibleMomentError,
     PhasedFSimCalibrationRequest,
     PhasedFSimCalibrationResult,
+    PhasedFSimCharacterization,
     WITHOUT_CHI_FLOQUET_PHASED_FSIM_CHARACTERIZATION,
+    ZETA_GAMMA_FLOQUET_PHASED_FSIM_CHARACTERIZATION,
     try_convert_sqrt_iswap_to_fsim,
 )
 from cirq.google.engine import Engine
 from cirq.google.serializable_gate_set import SerializableGateSet
 
 
+@dataclasses.dataclass(frozen=True)
+class CircuitCalibration:
+    """Circuit with calibration data annotations.
+
+    Attributes:
+        circuit: Circuit instance.
+        moment_allocations: Maps each moment within a circuit to an index of a characterization
+            request or response. None means that there is no characterization data for that moment.
+    """
+
+    circuit: Circuit
+    moment_allocations: List[Optional[int]]
```

Comment:
and this

Reply:
yes, allocation seems like the wrong word here. `moment_to_index`? `moment_to_record_index`? `moment_to_req[_index]`?

Reply:
I made it moment_to_calibration for consistency.